app crash when JWT is loaded from network

this happens if JWT stored in a network location (like NextCloud)

```
04-06 16:07:49.962 20599 20599 E AndroidRuntime: Caused by: android.os.NetworkOnMainThreadException
04-06 16:07:49.962 20599 20599 E AndroidRuntime: 	at android.os.Parcel.createExceptionOrNull(Parcel.java:3271)
04-06 16:07:49.962 20599 20599 E AndroidRuntime: 	at android.os.Parcel.createException(Parcel.java:3245)
04-06 16:07:49.962 20599 20599 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:3228)
04-06 16:07:49.962 20599 20599 E AndroidRuntime: 	at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:201)
04-06 16:07:49.962 20599 20599 E AndroidRuntime: 	at android.database.DatabaseUtils.readExceptionWithFileNotFoundExceptionFromParcel(DatabaseUtils.java:164)
04-06 16:07:49.962 20599 20599 E AndroidRuntime: 	at android.content.ContentProviderProxy.openTypedAssetFile(ContentProviderNative.java:814)
04-06 16:07:49.962 20599 20599 E AndroidRuntime: 	at android.content.ContentResolver.openTypedAssetFileDescriptor(ContentResolver.java:2043)
04-06 16:07:49.962 20599 20599 E AndroidRuntime: 	at android.content.ContentResolver.openAssetFileDescriptor(ContentResolver.java:1858)
04-06 16:07:49.962 20599 20599 E AndroidRuntime: 	at android.content.ContentResolver.openInputStream(ContentResolver.java:1528)
04-06 16:07:49.962 20599 20599 E AndroidRuntime: 	at org.openziti.mobile.ZitiEnrollmentActivity.enroll(ZitiEnrollmentActivity.kt:123)
04-06 16:07:49.962 20599 20599 E AndroidRuntime: 	at org.openziti.mobile.ZitiEnrollmentActivit
```

original [report](https://openziti.discourse.group/t/ziti-network-jwt-enrollment-fails-on-android-when-loaded-from-nextcloud/4310)

--- a/app/src/main/java/org/openziti/mobile/ZitiEnrollmentActivity.kt
+++ b/app/src/main/java/org/openziti/mobile/ZitiEnrollmentActivity.kt
@@ -10,6 +10,7 @@ import android.hardware.SensorManager
 import android.net.Uri
 import android.os.Bundle
 import android.os.Handler
+import android.util.Log
 import android.view.OrientationEventListener
 import android.widget.LinearLayout
 import android.widget.Toast
@@ -19,6 +20,9 @@ import androidx.appcompat.app.AppCompatActivity
 import com.google.zxing.BarcodeFormat
 import com.journeyapps.barcodescanner.ScanContract
 import com.journeyapps.barcodescanner.ScanOptions
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
 import org.openziti.mobile.databinding.ActivityZitiEnrollmentBinding
 
 
@@ -88,13 +92,13 @@ class ZitiEnrollmentActivity : AppCompatActivity() {
         orienter.enable()
     }
 
-    fun jwtEnroll(launcher: ActivityResultLauncher<Intent>) {
+    private fun jwtEnroll(launcher: ActivityResultLauncher<Intent>) {
         val intent = Intent(Intent.ACTION_GET_CONTENT)
         intent.type = "*/*"
         launcher.launch(intent)
     }
 
-    fun qrEnroll(scanLauncher: ActivityResultLauncher<ScanOptions>) {
+    private fun qrEnroll(scanLauncher: ActivityResultLauncher<ScanOptions>) {
         val scanOptions = ScanOptions().apply {
             setDesiredBarcodeFormats(BarcodeFormat.QR_CODE.name)
             setPrompt("Scan enrollment code")
@@ -104,9 +108,10 @@ class ZitiEnrollmentActivity : AppCompatActivity() {
         scanLauncher.launch(scanOptions)
     }
 
-    fun enroll(jwt: String) {
+    private fun enroll(jwt: String) {
         (application as ZitiMobileEdgeApp).model.enroll(jwt)
             .handleAsync { _, ex ->
+                Log.i(TAG, "enroll result $ex")
                 Handler(mainLooper).post {
                     this.finish()
                     if (ex != null) {
@@ -119,10 +124,11 @@ class ZitiEnrollmentActivity : AppCompatActivity() {
             }
     }
 
-    fun enroll(jwtUri: Uri) {
-        application.contentResolver.openInputStream(jwtUri)?.reader()?.use {
-            enroll(it.readText())
-            this.finish()
+    private fun enroll(jwtUri: Uri) {
+        CoroutineScope(Dispatchers.IO).launch {
+            application.contentResolver.openInputStream(jwtUri)?.reader()?.use {
+                enroll(it.readText())
+            }
         }
     }
 

